src/query/functions/src/aggregates/aggregate_unary.rs

```diff
@@ -0,0 +1,217 @@
+use common_expression::Column;
+use common_expression::ColumnBuilder;
+use common_expression::Scalar;
+use common_expression::StateAddr;
+
+use crate::aggregates::aggregate_function_factory::AggregateFunctionDescription;
+use crate::aggregates::aggregate_quantile_disc_v2::create_quantile_disc;
+use crate::aggregates::aggregate_skewness_v2::create_skewness;
+use crate::aggregates::assert_unary_arguments;
+
+pub trait UnaryState<T>: Send + Sync + Clone
+where T: ValueType
+{
+    fn add(&mut self, other: Option<T::ScalarRef<'_>>);
+    fn add_batch(&mut self, column: &T::Column, validity: Option<&Bitmap>) -> Result<()>;
+    fn merge(&mut self, rhs: &Self) -> Result<()>;
+    fn merge_result(&mut self, builder: &mut ColumnBuilder) -> Result<()>;
+
+    fn serialize(&self, writer: &mut Vec<u8>) -> Result<()>;
+
+    fn deserialize(&self, reader: &mut &[u8]) -> Result<Self>
+    where Self: Sized;
+
+    fn need_manual_drop_state() -> bool {
+        false
+    }
+}
+
+pub struct AggregateUnaryFunction<S, T>
+where
+    S: UnaryState<T>,
+    T: ValueType + ArgType,
+{
+    display_name: String,
+    state: S,
+    _params: Vec<Scalar>,
+    _argument: DataType,
+    return_type: DataType,
+    _phantom: PhantomData<T>,
+}
+
+impl<S, T> Display for AggregateUnaryFunction<S, T>
+where
+    S: UnaryState<T>,
+    T: ValueType + ArgType,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+impl<S, T> AggregateUnaryFunction<S, T>
+where
+    S: UnaryState<T> + 'static,
+    T: Send + Sync + ArgType + ValueType + 'static,
+{
+    pub(crate) fn try_create(
+        display_name: &str,
+        return_type: DataType,
+        _params: Vec<Scalar>,
+        _argument: DataType,
+        state: S,
+    ) -> Result<Arc<dyn AggregateFunction>> {
+        let func: AggregateUnaryFunction<S, T> = AggregateUnaryFunction {
+            display_name: display_name.to_string(),
+            return_type,
+            _params,
+            _argument,
+            state,
+            _phantom: Default::default(),
+        };
+
+        Ok(Arc::new(func))
+    }
+}
+
+impl<S, T> AggregateFunction for AggregateUnaryFunction<S, T>
+where
+    S: UnaryState<T> + 'static,
+    T: Send + Sync + ArgType + ValueType + 'static,
+{
+    fn name(&self) -> &str {
+        &self.display_name
+    }
+
+    fn return_type(&self) -> Result<DataType> {
+        Ok(self.return_type.clone())
+    }
+
+    fn init_state(&self, place: StateAddr) {
+        place.write_state(self.state.clone())
+    }
+
+    fn state_layout(&self) -> Layout {
+        Layout::new::<S>()
+    }
+
+    fn accumulate(
+        &self,
+        place: StateAddr,
+        columns: &[Column],
+        validity: Option<&Bitmap>,
+        _input_rows: usize,
+    ) -> Result<()> {
+        let state: &mut S = place.get::<S>();
+        let column = T::try_downcast_column(&columns[0]).unwrap();
+        state.add_batch(&column, validity)
```

Comment:
`add_batch`  does not to be a method of state, just use  `state.add(value);` in this function.